src/browser.ts

```diff
@@ -1,13 +1,30 @@
 import { FinchMessage } from "./types";
 
+/* 
+  NOTE: chrome apis need to return true the call sendResponse
+  https://developer.chrome.com/docs/extensions/mv2/messaging/#simple
+*/
+
 export const addMessageListener = (
   handler: (
     message: FinchMessage,
     sender: browser.runtime.MessageSender | chrome.runtime.MessageSender
-  ) => Promise<unknown>
+  ) => Promise<unknown>,
+  options: { messageKey: string }
 ) => {
```

Comment:
for these two parameters, `handler` and `options`, is it possible to use an interface or for something like this it's not worth the hassle?

Reply:
will def use an interface if we expand the options, but there is not an easy way to combine the two. Mostly not worrying about this since most of these browser polyfills are used in one place only.